api/storage_service.cc

```diff
@@ -668,23 +668,16 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         auto keyspace = validate_keyspace(ctx, req->param);
         auto table_infos = parse_table_infos(keyspace, ctx, req->query_parameters, "cf");
         apilog.debug("force_keyspace_compaction: keyspace={} tables={}", keyspace, table_infos);
+
+        auto& cm = db.local().get_compaction_manager();
+        auto compaction_module_ptr = cm.get_compaction_module().shared_from_this();
+        auto task_impl_ptr = std::make_unique<major_keyspace_compaction_task_impl>(compaction_module_ptr, std::move(keyspace), db, table_infos);
+        auto task = co_await compaction_module_ptr->make_task(std::move(task_impl_ptr), {});
```

Comment:
we could hide all this boiler plate into a function living in a file implementing compaction module. that doesn't have to be exposed into api/storage_service.cc

I think I have a big problem with the name get_compaction_module(). When I read "compaction_module", it means to me the compaction layer, not the task_manager_compaction_module, which is perhaps what it should be called.